Mac OS X Default Assembler Irreparably Broken

Mac OS X's default assembler has major issues in Intel syntax mode.  Specifically, backward jumps are broken.  The only relevant information that I could find is [here](https://discussions.apple.com/thread/2149747?tstart=0).

Additionally, the default assembler installed by `brew install binutils` does not include an assembler.

```
~/pwntools ❯❯❯ uname
Darwin
~/pwntools ❯❯❯ /usr/bin/as -v
Apple Inc version cctools-862, GNU assembler version 1.38
~/pwntools ❯❯❯ asm $'foo:\njmp foo'
[ERROR] There was an error running ['/usr/bin/as', '-arch', 'i386', '-o', '/var/folders/c8/3mf71lv51172mvh8yhzlpv1w008rd2/T/pwn-asm-guNHlD/step2', '/var/folders/c8/3mf71lv51172mvh8yhzlpv1w008rd2/T/pwn-asm-guNHlD/step1']:
    It had the exitcode 1.
    It had this on stdout:
    /var/folders/c8/3mf71lv51172mvh8yhzlpv1w008rd2/T/pwn-asm-guNHlD/step1:5:suffix or operands invalid for `jmp'

[ERROR] An error occurred while assembling:
    .org 0x0
    .section .shellcode,"ax"
    .intel_syntax noprefix
    foo:
    jmp foo
...
pwnlib.exception.PwnlibException: There was an error running ['/usr/bin/as', '-arch', 'i386', '-o', '/var/folders/c8/3mf71lv51172mvh8yhzlpv1w008rd2/T/pwn-asm-guNHlD/step2', '/var/folders/c8/3mf71lv51172mvh8yhzlpv1w008rd2/T/pwn-asm-guNHlD/step1']:
It had the exitcode 1.
It had this on stdout:
/var/folders/c8/3mf71lv51172mvh8yhzlpv1w008rd2/T/pwn-asm-guNHlD/step1:5:suffix or operands invalid for `jmp'
```

--- a/pwnlib/asm.py
+++ b/pwnlib/asm.py
@@ -124,11 +124,9 @@ def _assembler():
         'little': '-EL'
     }[context.endianness]
 
-    osx = (platform.system() == 'Darwin')
-
     assemblers = {
-        'i386'   : [gas, '--32'] if not osx else [gas, '-arch', 'i386'],
-        'amd64'  : [gas, '--64'] if not osx else [gas, '-arch', 'x86_64'],
+        'i386'   : [gas, '--32'],
+        'amd64'  : [gas, '--64'],
 
         # Most architectures accept -EL or -EB
         'thumb'  : [gas, '-mthumb', E],
@@ -148,7 +146,6 @@ def _assembler():
     }
 
     return assemblers.get(context.arch, [gas])
-
 
 def _objcopy():
     return [which_binutils('objcopy')]
@@ -335,10 +332,8 @@ def asm(shellcode, vma = 0, **kwargs):
     result = ''
 
     with context.local(**kwargs):
-        osx = (platform.system() == 'Darwin')
-
         assembler = _assembler()
-        objcopy   = _objcopy() + ['-j', '.shellcode' if not osx else '*.shellcode*', '-Obinary']
+        objcopy   = _objcopy() + ['-j', '.shellcode', '-Obinary']
         code      = '.org %#x\n' % vma
         code      += _arch_header()
         code      += cpp(shellcode)

